solr/core/src/java/org/apache/solr/search/facet/FacetFieldMerger.java

```diff
@@ -159,6 +162,50 @@ public Object getMergedResult() {
     return result;
   }
 
+  @Override
+  public Object getPrototype() {
+    return null;
+  }
+
+  @Override
+  public void readState(JavaBinCodec codec, DataInputInputStream dis, Context mcontext) throws IOException {
```

Comment:
I'd say the `Context` usage is probably the only gap with most of the facet module iterative bits I have added (outside of the one or two explicitly excluded facet aggregations). It stands to assign facet buckets unique ids and track which shards had which buckets. I'm pretty sure it is solely used for facet refinement - which we have never used. I think we could make it work. I doubt refinement would work as expected now, but fixing this feels like a follow up if necessary. It is otherwise only passed around for the purposes of being able to merge in data.

Need to do one more cross check here though..

Reply:
Did you do the cross check? This only affects facet refinement which we don't use?  Probably worth calling that out in a comment somewhere prominent

Reply:
Yep, confirmed the context is only truly important for refinements. Will call it out somewhere..